test-data/unit/fine-grained.test

```diff
@@ -8206,6 +8206,35 @@ x = 'no way'
 ==
 main:10: error: Incompatible types in assignment (expression has type "str", variable has type "int")
 
+[case testNamedTupleForwardFunctionDirect]
+# flags: --ignore-missing-imports
+from typing import NamedTuple
+from b import B
+
+NT = NamedTuple('NT', [('x', B)])
```

Comment:
Maybe also try revealing the type of the `x` attribute?

Reply:
I added separate test (since `reveal_type` is an error and affects how targets are processed). This reminded me about an old question/TODO in `typeanal.py`. We still keep `UnboundType`s in some places instead of replacing them with `Any`. Maybe we should stop doing this?(Note this requires updating many test cases.)